one/src/daemon.rs

```diff
@@ -164,12 +164,14 @@ pub struct DaemonOpts {
     )]
     authentication: bool,
 
-    /// Enable event validation; Requires using the experimental-features flag
+    /// Enable event validation
     #[arg(
         long,
-        default_value_t = false,
-        env = "CERAMIC_ONE_EVENT_VALIDATION",
-        requires = "experimental_features"
+        default_value = "true",
+        value_parser = clap::value_parser!(bool),
+        num_args = 0..=1,
+        action = ArgAction::Set,
+        env = "CERAMIC_ONE_EVENT_VALIDATION"
     )]
     event_validation: bool,
```

Comment:
Re booleans being weird in clap. I think all we need to do re the issue you linked is set the to `Option<bool>` and then default it to true in our code like, `opts.event_validation.unwrap_or(true)`.

Reply:
Yeah! we make it optional and don'e set a default value. Seems to work : 
```samikas@Samikas-MacBook-Pro rust-ceramic % RUST_LOG=info,ceramic_olap=debug cargo run --bin ceramic-one -- daemon | grep "Event service created"
warning: unused import: `ArgAction`
  --> one/src/daemon.rs:17:12
   |
17 | use clap::{ArgAction, Args};
   |            ^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: `ceramic-one` (lib) generated 1 warning (run `cargo fix --lib -p ceramic-one` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.51s
     Running `target/debug/ceramic-one daemon`
  2024-10-22T18:08:19.257286Z  INFO ceramic_one::daemon: Event service created, event_validation: true
^C
samikas@Samikas-MacBook-Pro rust-ceramic % RUST_LOG=info,ceramic_olap=debug cargo run --bin ceramic-one -- daemon --event-validation=false | grep "Event service created"
warning: unused import: `ArgAction`
  --> one/src/daemon.rs:17:12
   |
17 | use clap::{ArgAction, Args};
   |            ^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: `ceramic-one` (lib) generated 1 warning (run `cargo fix --lib -p ceramic-one` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.50s
     Running `target/debug/ceramic-one daemon --event-validation=false`
  2024-10-22T18:08:39.913484Z  INFO ceramic_one::daemon: Event service created, event_validation: false```

Reply:
Added setting the default value using default_value. Added reasoning in the commit message incase anyone is interested in the workings of default_value vs default_value_t, why one works and the other does not